packages/rspeedy/lynx-bundle-rslib-config/src/LazyBundleWebpackPlugin.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright 2025 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+import type { Asset, Compilation, Compiler } from 'webpack'
+
+interface LazyBundleWebpackPluginOptions {
+  templateFileName: string
+  encode: (opts: unknown) => Promise<{ buffer: Buffer }>
+}
+
+export class LazyBundleWebpackPlugin {
+  constructor(private options: LazyBundleWebpackPluginOptions) {}
+
+  apply(compiler: Compiler): void {
+    compiler.hooks.thisCompilation.tap(
+      LazyBundleWebpackPlugin.name,
+      (compilation) => {
+        compilation.hooks.processAssets.tapPromise(
+          {
+            name: LazyBundleWebpackPlugin.name,
+            stage:
+              /**
+               * Generate the html after minification and dev tooling is done
+               * and source-map is generated
+               */
+              compiler.webpack.Compilation
+                .PROCESS_ASSETS_STAGE_OPTIMIZE_HASH,
+          },
+          () => {
+            return this.#generateLazyBundle(
+              compiler,
+              compilation,
+            )
+          },
+        )
+      },
+    )
+  }
+
+  async #generateLazyBundle(
+    compiler: Compiler,
+    compilation: Compilation,
+  ): Promise<void> {
+    const assets = compilation.getAssets()
+    const { buffer } = await this.#encode(assets)
+
+    const { RawSource } = compiler.webpack.sources
+    compilation.emitAsset(
+      this.options.templateFileName,
+      new RawSource(buffer, false),
+    )
+  }
+
+  async #encode(assets: Readonly<Asset>[]) {
+    const customSections = assets
+      .filter(({ name }) => name.endsWith('.js'))
+      .reduce<Record<string, { content: string }>>((prev, cur) => ({
+        ...prev,
+        [cur.name.replace('.js', '')]: {
+          content: cur.source.source().toString(),
+        },
+      }), {})
+
+    const { buffer } = await this.options.encode({
+      compilerOptions: {
+        enableFiberArch: true,
+        useLepusNG: true,
+        enableReuseContext: true,
+        bundleModuleMode: 'ReturnByFunction',
+        targetSdkVersion: '2.11',
```

Comment:
Is it safe to hard code `compilerOptions` here? I remember loading lazy bundle which has different `enableParallelElement` config with host bundle will fail.